specs/kyber/src/sampling.rs

```diff
@@ -5,67 +5,78 @@ use crate::{
     BadRejectionSamplingRandomnessError,
 };
 
-/// Given a series of uniformly random bytes in `|randomness|`, sample uniformly
-/// at random a ring element rˆ, which is treated as being the NTT representation
-/// of a corresponding polynomial r.
+/// If `|bytes|` contains a set of uniformly random bytes, this function
+/// uniformly samples a ring element â that is treated as being the NTT representation
+/// of the corresponding polynomial a.
 ///
-/// This implementation uses rejection sampling; it is therefore possible the
-/// supplied randomness is not enough to sample the element, in which case
-/// an Err is returned and the caller must try again with fresh randomness.
+/// Since rejection sampling is used, it is possible the supplied bytes are
+/// not enough to sample the element, in which case an Err is returned and the
+/// caller must try again with a fresh set of bytes.
 ///
-/// This function implements Algorithm 1 of the Kyber Round 3 specification,
-/// which is reproduced below:
+/// This function partially implements Algorithm 6 of the NIST FIPS 203 standard,
+/// We say "partially" because this implementation only accepts a finite set of
+/// bytes as input and returns an error if the set is not enough; Algorithm 6 of
+/// the FIPS 203 standard on the other hand samples from an infinite stream of bytes
+/// until the ring element is filled. Algorithm 6 is reproduced below:
 ///
-/// ```plaintext
-/// Input: Byte stream B = b_0, b_1, b_2 ... ∈ B^{*}
-/// Output: NTT-representation aˆ ∈ R_q of a ∈ R_q
-/// i := 0
-/// j := 0
-/// while j < n do
-///     d_1 := b_i + 256·(b_{i+1} mod^{+}16)
-///     d_2 := ⌊b_{i+1}/16⌋ + 16·b_{i+2}
-///     if d_1 < q then
-///         aˆ_{j} := d_1
-///         j := j + 1
+/// Input: byte stream B ∈ B*
+/// Output: array â ∈ ℤ₂₅₆
+///
+/// i ← 0
+/// j ← 0
+/// while j < 256 do
+///     d₁ ← B[i] + 256·(B[i+1] mod 16) d2 ← ⌊B[i+1]/16⌋+16·B[i+2] ifd1 <qthen
```

Comment:
```suggestion
///     d₁ ← B[i] + 256·(B[i+1] mod 16) d2 ← ⌊B[i+1]/16⌋+16·B[i+2] if d1 < q then
```

Reply:
This was text that wasn't supposed to be there; I removed it.